vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc

```diff
@@ -57,7 +109,7 @@ ledger::ContributionQueuePtr FromStringToContributionQueue(
 
   auto* type = dictionary->FindKey("type");
   if (type && type->is_int()) {
-    queue->type = static_cast<ledger::RewardsType>(type->GetInt());
+    queue->type = ConvertIntToRewardsType(type->GetInt());
```

Comment:
The new unit tests pass in 'illegal' values for enum's (i.e. RewardsType) in sample JSON - and they were being cast to 'illegal' enum values which would trip up logic elsewhere. 

Instead we should be validating all enumerated types received from the server as `int` and handling accordingly. I'll raise a ticket to make sure code which uses these three enum types deals with `UNKNOWN` values.

Reply:
ah, you've answered my questions above :-)

Reply:
Raised https://github.com/brave/brave-browser/issues/7897 to cover the above